enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,206 @@
+  * Configs that involve file writing to immutable local file system directories need to be handled: The immutable CoreOS model guards the immutability of certain directories. With that being said, if we write a file (e.g. SSH key) to an immutable directory during an OS image build and use rpm-ostree to rebase onto the newly built OS image, that file will not actually be there when the node reboots - basically any path under */var* will be affected by this. In the interim, the MCO will continue to write these files manually onto the  nodes as it currently does. In the future, the plan is to move these files into */usr* or */etc* or use systemd-tmpfiles dropins instead.
+  * Address functionality gap between bootc and rpm-ostree: figure out a story for kernel argument changes as a container image layer/label
+    - There is an update in support for this by bootc here https://containers.github.io/bootc/building/kernel-arguments.html.
+    - The MCO team did a spike investigation on how to support kernel args with bootc and the findings and planned solution can be found in this [jira card](https://issues.redhat.com/browse/MCO-1244).
+  * Carve a path for changes made to nodes without applying MachineConfig like changes made by other OCP operators (e.g. SRIO-V) or MCO cert writing.
+
+### Phase 3: Deprecate rpm-ostree and Move Layered Image Update Path to Bootc 
+
+Upon completion of the above, the last step is to update the image rollout mechanism. In the foreseeable future, we will be expecting a dual-stream RHCOS where rpm-ostree will be gradually deprecated and transition its responsibility to bootc. Following this trend, MCO will also enable image deployment by both “rpm-ostree rebase” and “bootc switch” and, in the end, settle down on “bootc switch” for image rollout. 
+
+### Workflow Description
+
+1. The cluster administrator wants to add new configuration to the OS on each of their cluster nodes.
+2. The cluster administrator specifies base OS image pull spec & secret, base OS extension pull spec, rendered image pull spec & secret, rendered image push spec & secret and custom content through a MachineOSConfig object.
+3. The cluster administrator specifies machine configuration through MachineConfig object.
+4. The Machine Config Controller will read the MachineConfig, check reconcilability and render a rendered-MachineConfig. 
+5. The BuildController detects a build request from new MachineOSConfig/MachineConfig creation and creates a build. A new MachineOSBuild object will be created to track build progress and mark targeted rendered-MachineConfig.
+6. The built image will be pushed to the rendered image push spec specified in the MachineOSConfig object (Step 2).
+7. The MachineConfigDaemon will then drain the node and call “bootc switch” to roll the newly built image to the node, and reboot the node upon completion.
+8. If required, cluster administrator can roll back to previous image.
```

Comment:
Worth adding details into how the admin triggers a rollback.

Reply:
Fixed